File: solution_set1/775. Global and Local Inversions.py
class Solution:
    def isIdealPermutation(self, nums: List[int]) -> bool:
        
        ## all local inv are also global inv
        
        for x in range(len(nums)):
            if abs(nums[x]-x) > 1 :
                return False
            
        return True
        #return all(abs(nums[x]-x)<=1 for x in range(len(nums)))
        

File: cpp_solutions/775. Global and Local Inversions.cpp
class Solution {
public:
    bool isIdealPermutation(vector<int>& nums) {
        
        for(int x=0 ; x<nums.size();++x) {
            if (abs(nums[x]-x) > 1) {return false;}
        }
        return true;
        
    }
};
